tests/test_advection.py

```diff
@@ -364,3 +365,58 @@ def test_decaying_eddy(fieldset_decaying, mode, method, rtol, npart=1):
     exp_lat = [truth_decaying(x, y, endtime)[1] for x, y, in zip(lon, lat)]
     assert np.allclose(pset.lon, exp_lon, rtol=rtol)
     assert np.allclose(pset.lat, exp_lat, rtol=rtol)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_analyticalAgrid(mode):
+    lon = np.arange(0, 15, dtype=np.float32)
+    lat = np.arange(0, 15, dtype=np.float32)
+    U = np.ones((lat.size, lon.size), dtype=np.float32)
+    V = np.ones((lat.size, lon.size), dtype=np.float32)
+    fieldset = FieldSet.from_data({'U': U, 'V': V}, {'lon': lon, 'lat': lat}, mesh='flat')
+    pset = ParticleSet(fieldset, pclass=ptype[mode], lon=1, lat=1)
+    failed = False
+    try:
+        pset.execute(AdvectionAnalytical, runtime=1)
+    except NotImplementedError:
+        failed = True
+    assert failed
+
+
+@pytest.mark.parametrize('mode', ['scipy'])  # JIT not implemented
+@pytest.mark.parametrize('u', [1, -0.2, -0.3, 0])
+@pytest.mark.parametrize('v', [1, -0.3, 0, -1])
+@pytest.mark.parametrize('w', [None, 1, -0.3, 0, -1])
+@pytest.mark.parametrize('direction', [1, -1])
+def test_uniformanalytical(mode, u, v, w, direction, tmpdir):
```

Comment:
for readability and simplicity, could you rename `test_uniformanalytical` to `test_uniform_analytical` ? That would be nice.